pkg/controller/volume/persistentvolume/binder_test.go

```diff
@@ -767,6 +767,44 @@ func TestSyncAlphaBlockVolume(t *testing.T) {
 			withClaimVolumeMode(&modeFile, newClaimArray("claim14-12", "uid14-12", "10Gi", "volume14-12", v1.ClaimBound, nil, annBoundByController, annBindCompleted)),
 			noevents, noerrors, testSyncClaim,
 		},
+		{
+			// syncVolume output warning when pv is prebound to pvc with mismatching volumeMode
+			"14-13 - output warning when pv is prebound to pvc with different volumeModes",
+			withVolumeVolumeMode(&modeFile, newVolumeArray("volume14-13", "10Gi", "uid14-13", "claim14-13", v1.VolumePending, v1.PersistentVolumeReclaimRetain, classEmpty, annBoundByController)),
```

Comment:
I think we don't want the annBoundByController annotations set

Reply:
Thank you for your comment. I checked the codes again for annotations for dynamic provision case. 

- Internal provisioner case:
  annBoundByController and annDynamicallyProvisioned are set in below code.
    https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/persistentvolume/pv_controller.go#L1433
    https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/volume/persistentvolume/pv_controller.go#L1434

- External provisioner case:
  annDynamicallyProvisioned is set in below code.
    https://github.com/kubernetes-incubator/external-storage/blob/master/lib/controller/controller.go#L905


With above in mind, do you think that it's better to just remove annBoundByController for all the cases or add some cases that have different set of annotations. (In the latter case, annDynamicallyProvisioned might also need to be added.)

Above annotations doesn't affect any behaviours in the current codes.
Therefore, I'm inclined to agree to just remove it, however let me confirm it just to be sure.

Reply:
Oh nevermind, I got the ordering wrong.  They are put on the PV after dynamic provisioning, and the new event generated here is after that point.  So this is correct.